embed_grader/serapis/views/tasks.py

```diff
@@ -149,3 +151,42 @@ def zip_input_files(request, task_id):
     response["Content-Disposition"] = "attachment; filename={0}_{1}_input.zip".format(assignment, task)
 
     return response
+
+
+@login_required(login_url='/login/')
+def view_task_input_files(request, task_id):
+    try:
+        task = AssignmentTask.objects.get(id=task_id)
+    except AssignmentTask.DoesNotExist:
+        return HttpResponse("Assignment task cannot be found")
+
+    user = User.objects.get(username=request.user)
+    assignment = task.assignment_fk
+    course = assignment.course_fk
+    if not user.has_perm('view_assignment', course):
```

Comment:
Did you also check if the assignment is not released yet, or if the task mode feedback/hidden so students shouldn't see the task?

You may want to check `AssignmentTask` model, it may have some helper methods

Reply:
If the task mode is feedback/hidden or if the assignment is not released yet, students can still use the url to open the visualization page, but no content is showing up. I've changed to show "not enough privilege" message as that might make more sense.

The permission checking logic is in https://github.com/nesl/AutoGrader/blob/e8f815e844b7f9b62212285b73ab11afa2b94a6c/embed_grader/serapis/models.py#L291.

Reply:
Gotcha!